cirq-core/cirq/transformers/routing/mapping_manager.py

```diff
@@ -130,6 +130,31 @@ def shortest_path(self, lq1: 'cirq.Qid', lq2: 'cirq.Qid') -> Sequence['cirq.Qid'
         physical_shortest_path = self._physical_shortest_path(self._map[lq1], self._map[lq2])
         return [self._inverse_map[pq] for pq in physical_shortest_path]
 
-    @cached_method
+    @_compat.cached_method
     def _physical_shortest_path(self, pq1: 'cirq.Qid', pq2: 'cirq.Qid') -> Sequence['cirq.Qid']:
         return nx.shortest_path(self._induced_subgraph, pq1, pq2)
+
+    def _value_equality_values_(self):
+        if nx.is_directed(self.device_graph):
+            graph_equality = (
+                tuple(sorted(self.device_graph.nodes)),
+                tuple(sorted(self.device_graph.edges)),
+                True,
+            )
+        else:
+            graph_equality = (
+                tuple(sorted(self.device_graph.nodes)),
+                tuple(sorted(tuple(sorted(edge)) for edge in self.device_graph.edges)),
+                False,
+            )
+        map_equality = tuple(sorted(self._map.items()))
```

Comment:
Made those changes to make MappingManager serializable but kept the same graph equality for value checking since `nx.readwrite.json_graph.node_link_data(device_graph)` produces two different strings for two same undirected graphs with different edge insertion order.

Reply:
As mentioned earlier, can we use do the following for value_equality_values?

https://github.com/quantumlib/Cirq/blob/4a64b1e8752d95009194f46e168c9a0af30406f4/cirq-core/cirq/devices/device.py#L136-L142

Reply:
This doesn't work in the case we have two undirected graphs with different edge insertion order.